Jenkinsfile

```diff
@@ -157,128 +287,173 @@ pipeline {
-                    }
-                    agent {
-                        // 4 node cluster with 2 IB/node + 1 test control node
-                        label params.CI_NVME_5_LABEL
-                    }
-                    steps {
-                        // Need to get back onto base_branch for ci/
-                        checkoutScm url: 'https://github.com/daos-stack/daos.git',
-                                    branch: env.BaseBranch,
-                                    withSubmodules: true
-                        functionalTest inst_repos: daosRepos(),
-                                       inst_rpms: functionalPackages(1, next_version, "tests-internal"),
-                                       test_function: 'runTestFunctionalV2'
-                    }
-                    post {
-                        always {
-                            functionalTestPostV2()
-                        }
-                    }
-                } // stage('Functional_Hardware_Medium')
-                stage('Functional Hardware Large') {
-                    when {
-                        beforeAgent true
-                        expression { !skipStage() }
-                    }
-                    agent {
-                        // 8+ node cluster with 1 IB/node + 1 test control node
-                        label params.CI_NVME_9_LABEL
-                    }
-                    steps {
-                        // Need to get back onto base_branch for ci/
-                        checkoutScm url: 'https://github.com/daos-stack/daos.git',
-                                    branch: env.BaseBranch,
-                                    withSubmodules: true
-                        functionalTest inst_repos: daosRepos(),
-                                       inst_rpms: functionalPackages(1, next_version, "tests-internal"),
-                                       test_function: 'runTestFunctionalV2'
-                    }
-                    post {
-                        always {
-                            functionalTestPostV2()
-                        }
-                    }
-                } // stage('Functional_Hardware_Large')
-            } // parallel
+            steps {
+                script {
+                    // Note: The functional test steps define 'nvme' instead of 'default_nvme' to
+                    // force the launch.py --nvme argument.  This means the 'Test-nvme' commit
+                    // pragmas will be ignored. This is to avoid multiple parallel test stages
+                    // from duplicating testing.
+                    parallel(
+                        'Functional on EL 8': getFunctionalTestStage(
+                            name: 'Functional on EL 8',
+                            pragma_suffix: '-vm',
+                            distro: 'el8',
+                            base_branch: env.BaseBranch,
+                            label: vm9_label('EL8'),
```

Comment:
A bit of an aside but this feels duplicative of setting `distro:` as we do on line 300.  At least it could default perhaps?

Reply:
In the case of running a functional test stage that is distro specific, yes, it is somewhat redundant. In a functional test stage where the label doesn't reference the distro - like HW Medium - it isn't.  I have thought of explicitly specifying the `distro` argument to the Functional HW stage definitions.  If it was combined to a new build parameter, we could do more extensive testing of other distros easily.